### PR TITLE
Simplify local development

### PR DESCRIPTION
**Pull request by nstapelbroek**
Hey :raising_hand_man: 

This setup should allow anyone to develop on their local machine without using a "hosted" environment.
Could you take a look and see if this works on your machine?

### What has been changed:
- Added an attempt to handle SIGINT better, since this produces problems on a non-interactive container (It has something to do with `/bin/sh -C`  as CMD and node running on PID 1. I'm not entirely sure what causes this behavior though)
- Installed nodemon to restart the application when developing
- Created a docker-compose file so you can easely develop locally using ngrok and the application code in a mount. 

### How to test:
- Follow the readme :smile:

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+version: '2'
+
+services:
+    application:
+      build:
+          dockerfile: ./Dockerfile
+          context: .
+      command: yarn watch
+      environment:
+          RAZOR_API_KEY: pleasechangeme
```
Comment (daanpeer) [inline docker-compose.yml]: Maybe we can use the .env for the razor key :) 

https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option

File: README.MD [hunk 2]
```diff
@@ -2,20 +2,29 @@
...
+- A TextRazor API key
 - A Slack channel you can develop on
 
-## deployment
+Once you have these requirements in place, follow these steps to get the environment working:
+1. insert your TextRazor API key in the docker-compse.yml file as a envinroment variable for the application
+2. run `docker-compse up`
+3. navigate to http://localhost:8888 to view the ngrok status, use the given *.ngrok.io URL on this page to develop your application
+4. configure this ngrok url as your Slack APP url at https://api.slack.com/apps
+5. Make sure your configure your "Event Subscriptions" in the Slack applicaiton. Currently, the should I read this application requires at least [link_shared](https://api.slack.com/events/link_shared) and [message.channels](https://api.slack.com/events/message.channels)
```
Comment (daanpeer) [inline README.MD]: applicaiton = application 😛

File: src/helpers/signalHandler.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        'SIGTERM'
+    ].forEach((signal) => {
+        process.on(signal, () => {
+            console.log(`Quitting... Reason: ${signal}`);
+            process.exit();
+        });
+    });
+}
+
+module.exports = signalHandler;
```
Comment (daanpeer) [inline src/helpers/signalHandler.js]: Missing newline

File: README.MD [hunk 4]
```diff
@@ -2,20 +2,29 @@
...
 
-### containerize your own changes
+### Shipping your own changes
 You can use the `make build` to create an image and push it to docker hub. Use an environment variable to overwrite the registyrname like so:
 ```
 make build REGISTRYNAME=myawesomeregistry.com/username/should-i-read-this
-```
\ No newline at end of file
+```
+Lateron, you can pull the image from your custom registry and host it just like the "Deployment" describes.
```
Comment (daanpeer) [inline README.MD]: Just a space here

**Review by daanpeer — CHANGES_REQUESTED** [review]
For some reason I get a 502. It looks like the application hostname cannot be found from Ngrok. Do you know how we can resolve this?

File: docker-compose.yml (hunk 1 above)
Comment (nstapelbroek) [reply docker-compose.yml]: oh sweet! didn't know docker-compose supported a .env file. I'll create a .env.dist and ignore a .env which will be used in the docker-compose :+1:

**Comment by nstapelbroek** [comment]
@daanpeer Thanks for the feedback :beers: 
I've updated the PR

File: package.json [hunk 5]
```diff
@@ -5,13 +5,17 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node index.js"
+    "start": "node index.js",
+    "watch": "./node_modules/nodemon/bin/nodemon.js -L index.js"
```
Comment (daanpeer) [inline package.json]: You can just use `nodemon -L index.js` node will use the local node_modules first :)

**Comment by daanpeer** [comment]
The issue probably has to do with the node_env (nodemon isn't starting ;)) should be an easy fix

**Comment by daanpeer** [comment]
It does work 🎉

**Comment by nstapelbroek** [comment]
@daanpeer did changing the `nodemon` fix the entire setup? if so: I've pushed your requested change, lets merge!

**Pull request merged**